### ISSUE TITLE
Changing colour scale in FeaturePlot question

### ISSUE THREAD

**Issue by adelynechan** [satijalab/seurat#1721]
Hi authors, I'm trying to generate tSNE plots coloured by expression level of a gene of interest, using a custom colour palette which I specified by inputting HEX values into a vector:
> custom_colours <- c("#ffff33", "#fed976", "#ffd700", "#feb24c", "#fe9929", "#fd8d3c", "#ff7f00", "#f16913", "#ec7014", "#d94801", "#fc4e2a", "#e31a1c", "#e41a1c", "#bd0026", "#800026", "#7f0000", "#67000d", "#000000")

I then defined the feature to plot as a variable (to be used in a loop for multiple genes):
> current_gene <- "GAPDH"

Then called the following to save the resulting plot to a PDF. 
> pdf(paste0("~/Desktop/", current_gene, "_customColours.pdf"))
> FeaturePlot(seurat_object, reduction="tsne", features=c(current_gene), pt.size=2, cols=custom_colours)
> dev.off()

I made a bunch of these and was slightly surprised, as regardless of whether or not I expected my gene to be a high or low expressor, the markings on the scale bar remained the same. I attach two examples: GAPDH which should be highly expressed and CCNA1 which is tissue specific and not expected to be expressed in the cell type that I am using. 
[CCNA1_customColours.pdf](https://github.com/satijalab/seurat/files/3311957/CCNA1_customColours.pdf)
[GAPDH_customColours.pdf](https://github.com/satijalab/seurat/files/3311958/GAPDH_customColours.pdf)

I thought it was slightly odd that in both cases, the scale bar was the same (going all the way up to above 15), and in the CCNA1 plot there are a small number of cells which seem to express the gene to incredibly high levels. I then tried exactly the same command but without specifying the colour:
> pdf(paste0("~/Desktop/", current_gene, "_defaultColours.pdf"))
> FeaturePlot(seurat_object, reduction="tsne", features=c(current_gene), pt.size=2)
> dev.off()
[CCNA1_defaultColours.pdf](https://github.com/satijalab/seurat/files/3311967/CCNA1_defaultColours.pdf)
[GAPDH_defaultColours.pdf](https://github.com/satijalab/seurat/files/3311969/GAPDH_defaultColours.pdf)

and this time the scale bars seem to be totally different, and much closer to what I'd expect. Could you please advise how I can manually specify colours, but still have the correct values reflected in the legend? The points look like they are colouring correctly (in terms of high/low).

Thanks in advance!

Adelyne

**Comment by samuel-marsh** [comment]
Hi,
_Disclosure: Not part of the Seurat team._

The reasoning can be found in the Seurat documentation for FeaturePlot() call.
`Library(Seurat) / 
?FeaturePlot()`

for the "cols" argument it states: _The two colors to form the gradient over. Provide as string vector with the first color corresponding to low values, the second to high. Also accepts a Brewer color scale or vector of colors. Note: this will bin the data into number of colors provided._

So basically as you supplied 18 color list for the palette in created 18 bins (hence the scale going over 15).  See also issue I asked in #1540 with response from Satija Lab.

**Comment by adelynechan** [comment]
@samuel-marsh thanks for the info. I misunderstood what the documentation states (assumed it meant it would still be setting the max and min to the specific gene, and then making more bins i.e. breaks closer together with a bigger colour palette). If that were the case though, the output is quite unintuitive: In that case I'd expect the low expressor genes to all have a colour close to yellow (low in my colour palette), and only reaching the top end of the scale in genes with a large dynamic range.

**Comment by samuel-marsh** [comment]
Hi @vwtlin @AleksZakirov and others,

In time since this answer was provided I have also released my own R package to help with this issue (https://samuel-marsh.github.io/scCustomize/).  The function `FeaturePlot_scCustom` allows for plotting of custom gradients without the need for post-plot theming and adds a number of other quality of life features to the resulting plots (https://samuel-marsh.github.io/scCustomize/articles/Gene_Expression_Plotting.html#featureplot_sccustom-solves-these-issues).

The newest version (v1.0.0) should be available later today via GitHub or you can download the develop branch for most up to date version with bug fixes.

Best,
Sam

**Comment by DC-Jade** [comment]
> @adelynechan and @samuel-marsh and pretty much everybody else who will face the same problem
> 
> The perfect solution for me turned out to be using the `&` operator after `FeaturePlot()` to provide with a continuous color scale, e.g:
> 
> `FeaturePlot(seurat_object, reduction="tsne", features=c(current_gene), pt.size=2) & scale_color_viridis_c()`

great! Thanks
